src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -828,6 +879,185 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
+                        try
+                        {
+                            using (var requestExecutor =
+                                   ClusterRequestExecutor.CreateForSingleNode(nodeUrl, Server.Certificate.Certificate, DocumentConventions.DefaultForServer))
+                            using (requestExecutor.ContextPool.AllocateOperationContext(out var context))
+                            {
+                                try
+                                {
+                                    if (databaseNodes.Contains(tag)) // if node is part of db topology - wait for commit and database notification
+                                    {
+                                        var cmd = new WaitForDatabaseIndexNotificationCommand(index, databaseName);
+                                        await requestExecutor.ExecuteAsync(cmd, context, token: token);
+                                    }
+                                    else // else - wait for commit only
+                                    {
+                                        var cmd = new WaitForCommitCommand(index);
+                                        await requestExecutor.ExecuteAsync(cmd, context, token: token);
+                                    }
+                                }
+                                catch (Exception ex) when (ex is DatabaseDoesNotExistException or
+                                                               DatabaseNotRelevantException or
+                                                               DatabaseDisabledException or
+                                                               HttpRequestException)
+                                {
+                                }
+                            }
+                        }
+                        catch (RavenException ex) when (ex.InnerException is HttpRequestException)
+                        {
+                        }
+                        catch (Exception e) when (e is InvalidOperationException or AllTopologyNodesDownException)
+                        {
+                            // from request executor (ThrowFailedToContactAllNodes)
+                        }
+
+                    }
+                }
+
+            }
+            catch (OperationCanceledException)
+            {
+            }
+        }
+
+        public class WaitForDatabaseIndexNotificationCommand : RavenCommand
+        {
+            private readonly long _index;
+            private readonly string _database;
+
+            public WaitForDatabaseIndexNotificationCommand(long index, string database)
+            {
+                _index = index;
+                _database = database;
+            }
+
+            public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
+            {
+                url = $"{node.Url}/databases/{_database}/admin/rachis/wait-for-index-notification?index={_index}";
```

Comment:
this is database specific, for our case we need it on the server level

Reply:
Do you want to pass the database name as a parameter?